### PR TITLE
Tokio 0.2 blog post

### PR DESCRIPTION
**Pull request by carllerche**

### CODE REVIEW

File: content/blog/2019-11-tokio-0-2.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+```
+
+[API docs]: https://docs.rs/tokio/0.2
+
+This is a major update, and as such there are changes to almost all parts of the
+library. To highlight a few:
+
+* Based on `async / await` for superior ergonomics.
+
+* A brand new, much faster, scheduler.
```
Comment (hawkw) [inline content/blog/2019-11-tokio-0-2.md]: Maybe link to the scheduler blog post?

File: content/blog/2019-11-tokio-0-2.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+As of Rust 1.39, the `async / await` is available on the stable channel and
+Tokio 0.2 takes full advantage of it. Whenever possible, Tokio provides an
+`async fn` based API, usually modeled after `std`.
+
+For example, accepting sockets from a `TcpListener` is done with the async
+`accept` function:
+
+```rust,ignore
+let (mut socket, _) = listener.accept().await?;
```
Comment (seanmonstar) [inline content/blog/2019-11-tokio-0-2.md]:
```suggestion
let (mut socket, _peer_addr) = listener.accept().await?;
```

Just to clarify what the value is to anyone unfamiliar.

File: content/blog/2019-11-tokio-0-2.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+tokio = { version = "0.2", features = ["full"] }
+```
+
+On my laptop, this reduces the total number of crates to 23. Compile time only
+drops to 40 seconds.
+
+Real benefits start happening when the user of Tokio starts only requesting the
+components that are needed to run the application. To run the TCP echo server
+example, the `io-util`, `rt-threaded`, and `tcp` feature flags are needed. Now,
+Tokio pulls in 13 crates and compiling takes **13 seconds**.
```
Comment (hawkw) [inline content/blog/2019-11-tokio-0-2.md]:
```suggestion
Tokio pulls in 13 crates and compiles in **13 seconds**.
```

File: content/blog/2019-11-tokio-0-2.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+All that being said, we are shipping version 0.2. The question of "why not 1.0?"
+has come up a few times. After all, Tokio 0.1 has been stable for three years.
+The short answer: because it isn't time. There is nobody who would rather ship a
+Tokio 1.0 than us. It also isn't something to rush.
+
+After all, `async / await` only landed in the stable Rust channel weeks ago.
+There has been no significant production validation yet, except maybe fuchsia
+and that seems like a fairly specialized use case. This release of Tokio
+include significant new code and new strategies with feature flags. Also, there
```
Comment (seanmonstar) [inline content/blog/2019-11-tokio-0-2.md]:
```suggestion
includes significant new code and new strategies with feature flags. Also, there
```

File: content/blog/2019-11-tokio-0-2.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+There are still improvements to make, but initial testing using [Hyper] shows a
+30+% speed up in macro level benchmarks between the old scheduler and the new
+scheduler.
+
+You can read more about it [here][sched].
+
+Since it landed on master, some Tokio users have been experimenting with the new
+scheduler and have seen some very impressive real world improvements in their
+applications. Hopefully they will blog about them soon!
+
```
Comment (seanmonstar) [inline content/blog/2019-11-tokio-0-2.md]:
```suggestion

[Hyper]: https://hyper.rs
```

File: content/blog/2019-11-tokio-0-2.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+The Tokio 1.0 release will be **no later** than Q3 2020. It will also come with
+"long-term support" guarantees:
+
+* A minimum of 5 years of maintenance.
+* A minimum of 3 years before a hypothetical 2.0 release.
+
+When Tokio 1.0 is released in Q3 2020, on-going support, security fixes, and
+critical bug fixes are guaranteed until **at least** Q3 2025. Tokio 2.0 will not
+be released until **at least** Q3 2023 (though, ideally there will never been a
```
Comment (hawkw) [inline content/blog/2019-11-tokio-0-2.md]:
```suggestion
be released until **at least** Q3 2023 (though, ideally, there will never need to be a
```

**Pull request merged**